File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Persistence/database/OrganizatorRepository.cs
using System.Data;
using Model;
using log4net;

namespace Persistence.database;
public class OrganizatorRepository : AbstractRepository<int, Organizator>, IOrganizatorRepo
{
        private static readonly ILog logger = LogManager.GetLogger(typeof(OrganizatorRepository));
        private readonly IDbConnection connection;
        
        public OrganizatorRepository()
        {
            logger.Info("Initializing OrganizatorRepository...");
            connection = JdbcUtils.GetInstance().GetConnection();
        }

        protected override IDbCommand FindOneQuery(int id)
        {
            var command = connection.CreateCommand();
            command.CommandText = "SELECT * FROM organizatori WHERE org_id = @id";
            var param = command.CreateParameter();
            param.ParameterName = "@id";
            param.Value = id;
            command.Parameters.Add(param);
            return command;
        }

        protected override IDbCommand FindAllQuery()
        {
            var command = connection.CreateCommand();
            command.CommandText = "SELECT * FROM organizatori";
            return command;
        }

        protected override IDbCommand SaveQuery(Organizator entity)
        {
            var command = connection.CreateCommand();
            command.CommandText = "INSERT INTO organizatori (org_id, username, password) VALUES (@id, @username, @password)";
            
            var paramId = command.CreateParameter();
            paramId.ParameterName = "@id";
            paramId.Value = entity.Id;
            command.Parameters.Add(paramId);
            
            var paramUsername = command.CreateParameter();
            paramUsername.ParameterName = "@username";
            paramUsername.Value = entity.Username;
            command.Parameters.Add(paramUsername);
            
            var paramPassword = command.CreateParameter();
            paramPassword.ParameterName = "@password";
            paramPassword.Value = entity.Password;
            command.Parameters.Add(paramPassword);
            
            return command;
        }

        protected override Organizator BuildEntity(IDataReader reader)
        {
            var id = reader.GetInt32(reader.GetOrdinal("org_id"));
            var username = reader.GetString(reader.GetOrdinal("username"));
            var password = reader.GetString(reader.GetOrdinal("password"));
            
            return new Organizator(username, password) { Id = id };
        }
}

File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectPersistance/src/main/java/ro/mpp2025/javaprojectui/database/OrganizatorRepository.java
//package ro.mpp2025.javaprojectui.database;
//
//import org.apache.logging.log4j.LogManager;
//import org.apache.logging.log4j.Logger;
//import ro.mpp2025.javaprojectui.AbstractRepository;
//import ro.mpp2025.javaprojectui.Organizator;
//import ro.mpp2025.javaprojectui.interfaces.IOrganizatorRepository;
//import ro.mpp2025.javaprojectui.utils.JdbcUtils;
//
//
//import java.sql.Connection;
//import java.sql.PreparedStatement;
//import java.sql.ResultSet;
//import java.sql.SQLException;
//
//public class OrganizatorRepository extends AbstractRepository<Integer, Organizator> implements IOrganizatorRepository {
//
//    private static final Logger logger= LogManager.getLogger(OrganizatorRepository.class);
//
//    private final Connection connection;
//
//    public OrganizatorRepository() throws SQLException {
//        logger.info("Initializing OrganizatorRepository");
//        this.connection = JdbcUtils.getInstance().getConnection();
//    }
//
//    @Override
//    protected PreparedStatement findOneQuery(Integer id) throws SQLException {
//        String query = "SELECT * FROM organizatori WHERE id = ?";
//        PreparedStatement statement = connection.prepareStatement(query);
//        statement.setObject(1, id);
//
//        return statement;
//    }
//
//    @Override
//    protected PreparedStatement findAllQuery() throws SQLException {
//        String query = "SELECT * FROM organizatori";
//        return connection.prepareStatement(query);
//    }
//
//    @Override
//    protected PreparedStatement saveQuery(Organizator entity) throws SQLException {
//        String query = "INSERT INTO organizatori(id, username, password) VALUES (?, ?, ?)";
//        PreparedStatement statement = connection.prepareStatement(query);
//        statement.setObject(1, entity.getId());
//        statement.setString(2, entity.getUsername());
//        statement.setString(3, entity.getPassword());
//        return statement;
//    }
//
//    @Override
//    protected Organizator buildEntity(ResultSet resultSet) throws SQLException {
//        Integer id = resultSet.getInt("id");
//        String username = resultSet.getString("username");
//        String password= resultSet.getString("password");
//
//        Organizator organizator = new Organizator(username, password);
//        organizator.setId(id);
//
//        return organizator;
//    }
//
//}
